### DISCUSSION TITLE
Livewire image preview upload not showing

### DISCUSSION THREAD

**Discussion by worse435** [livewire/livewire#4238] [7. Bugs (archived)]
When i upload an image with livewire, image preview not showing.
Temporary url is generated correctly, but image don´t show. If I  copy the link of the image and i paste on my web browser it charge correctly but the image doesn´t show.

It fails on localhost and on a server.
I check config/livewire and i have a jpg option:
      'preview_mimes' => [   // Supported file types for temporary pre-signed file URLs.
            'png', 'jpg', 'jpeg',  'gif', 'bmp', 'svg', 'wav', 'mp4',
            'mov', 'avi', 'wmv', 'mp3', 'm4a',
            'jpeg', 'mpga', 'webp', 'wma',
        ],

I user:
"laravel/framework": "^8.12",
"livewire/livewire": "^2.3",
"php": "^7.3|^8.0",

can somebody help me?
Thanks

![image_preview](https://user-images.githubusercontent.com/44668528/142270777-900cfc91-1edc-4368-abd8-ce8b3f8ccef9.png)

![copy_url_image_to_browser](https://user-images.githubusercontent.com/44668528/142270782-e66dbead-524d-41dc-af2e-46a98e7e8da2.png)

**Comment by joshhanley** [comment]
@worse435 please share your component class and blade view.

When you paste code into github, you need to surround it in triple back ticks and then specify the file type like php or blade.

For example
\```php
\\ PHP code here
\```

\```blade
\\ HTML and blade code here
\```

Thanks!

**Comment by worse435** [comment]
```php
\ PHP code here
// livewire
use Livewire\Component;
use Livewire\WithPagination;
use Livewire\WithFileUploads;

class FormComponent extends Component
{

   use WithPagination;
   use WithFileUploads;
```

```blade
\ HTML and blade code here
 @if (!$foto)
      <div class="z-depth-1-half mb-2">
         <img src="{{asset('storage/images/varias/img_nodisponible.png')}}" class="" style="width:100%;height:100px;">
      </div>
   @elseif($foto && !is_string($foto))
      <div class="z-depth-1-half mb-2">
         <img src="{{$foto->temporaryUrl()}}" class="" style="width:100%;height:100px;">
      </div>
   @elseif ($selected_id)
      <div class="z-depth-1-half mb-2">
         <img src="{{$foto}}" class="" style="width:100%;height:100px;">
      </div>
   @endif
```

**Reply by worse435** [reply]
I have the same code in other projects and it works well. But in others not working. I dont now whats happened

**Reply by joshhanley** [reply]
@worse435 maybe when you upload the image, inspect the `img` tag and see what the src is, and see if you can browse to it manually. Maybe it's not linked publicly properly?

**Reply by worse435** [reply]
> @worse435 maybe when you upload the image, inspect the `img` tag and see what the src is, and see if you can browse to it manually. Maybe it's not linked publicly properly?

I browse it manually without error, but de image not load, load like this
![copy_url_image_to_browser](https://user-images.githubusercontent.com/44668528/142282219-43fa5aea-18ca-44e5-ac31-25851751fcc5.png)

**Reply by joshhanley** [reply]
@worse435 oh sorry, you did that say in your original message but I missed it!

Hmm hard to say without seeing it running.

Any chance you could create a small example repo which reproduces the issue with as little code as possible so we can investigate it?

**Reply by worse435** [reply]
I can´t create a small sample.
I think that the issue is in the route, because if I load the url image with a path/storage helper, the image load well. Only fail with temporary url helper.

**Reply by joshhanley** [reply]
@worse435 I'm sorry, without a small example repo I'm not able to be of much help, other than to advise to do what you are saying, which is to check all your routes, generated URL paths, storage config, etc, to make sure they are all right.

I've tested locally and I don't have any issues generating a temporary URL, so without seeing a working example of the problem I don't know how else I can help.

**Reply by worse435** [reply]
Thanks for your help anyway.
I solved with a temporary solution changing temporary url by storage url:
First, I change temporary_file_upload disk to public
```php
\ PHP code here
    'temporary_file_upload' => [
        'disk' => 'public', 
```
And then, I write url image manually with storage asset and filename
```blade
\ HTML and blade code here
  <img src="{{asset('storage').'/livewire-tmp/'.$foto->getFilename()}}" class="" style="width:100%;height:100px;">
```

**Reply by joshhanley** [reply]
@worse435 no worries! 🙂

Oh interesting, so when you compare what the `->temporaryUrl()` outputs vs what your manual URL outputs, is there any differences between them?

If so can you share a sample of each here?

**Reply by worse435** [reply]
Here the links:

With manual solution (Working):
http://localhost/XXX/htdocs/XXX/public/storage/livewire-tmp/K6S95BlG5u3A5vy4zRqpBaHrXLSYok-metaQ2FwdHVyYSBkZSBwYW50YWxsYSAyMDIxLTExLTE2IGEgbGFzIDE0LjQwLjEwLnBuZw==-.png

With TemporaryUrl (Not Working):
http://localhost/XXX/htdocs/XXX/public/livewire/preview-file/K6S95BlG5u3A5vy4zRqpBaHrXLSYok-metaQ2FwdHVyYSBkZSBwYW50YWxsYSAyMDIxLTExLTE2IGEgbGFzIDE0LjQwLjEwLnBuZw==-.png?expires=1637268757&signature=ebc7663dda7603a5c46f9149ba893d57887fc96e371e58d40df0d2c490dd801f

The different is the route and the signature and expires parameters

**Reply by joshhanley** [reply]
@worse435 yeah it seems there is an issue with what storage directory it is using, your manual one is using `public/storage/livewire-tmp` but the generated one is using `public/livewire/preview-file`.

Have you published and changed your Livewire config for `temporary_file_upload`? If so can you share it here?

**Reply by worse435** [reply]
It fails in both configurations.

Now (to solve the issue with storage asset)
```php
\ PHP code here
  'temporary_file_upload' => [
        'disk' => 'public',        // Example: 'local', 's3'              Default: 'default'
        'rules' => null,       // Example: ['file', 'mimes:png,jpg']  Default: ['required', 'file', 'max:12288'] (12MB)
        'directory' => null,   // Example: 'tmp'                      Default  'livewire-tmp'
        'middleware' => null,  // Example: 'throttle:5,1'             Default: 'throttle:60,1'
        'preview_mimes' => [   // Supported file types for temporary pre-signed file URLs.
            'png', 'jpg', 'jpeg',  'gif', 'bmp', 'svg', 'wav', 'mp4',
            'mov', 'avi', 'wmv', 'mp3', 'm4a',
            'jpeg', 'mpga', 'webp', 'wma',
        ],
    ],
```

After:
```php
\ PHP code here
  'temporary_file_upload' => [
        'disk' => null,        // Example: 'local', 's3'              Default: 'default'
        'rules' => null,       // Example: ['file', 'mimes:png,jpg']  Default: ['required', 'file', 'max:12288'] (12MB)
        'directory' => null,   // Example: 'tmp'                      Default  'livewire-tmp'
        'middleware' => null,  // Example: 'throttle:5,1'             Default: 'throttle:60,1'
        'preview_mimes' => [   // Supported file types for temporary pre-signed file URLs.
            'png', 'jpg', 'jpeg',  'gif', 'bmp', 'svg', 'wav', 'mp4',
            'mov', 'avi', 'wmv', 'mp3', 'm4a',
            'jpeg', 'mpga', 'webp', 'wma',
        ],
    ],
```

**Reply by joshhanley** [reply]
@worse435 ah bugger. Do you know where this url path `public/livewire/preview-file` would be coming from? Did you also change the directory or something?

But yeah that's where I'd be digging around is checking Livewire's config matches your filesystem config, and that the generated url is returning what is expected.

**Reply by VenomXL** [reply]
@joshhanley @worse435 I am facing exactly the same issue. Is there a solution to resolve this?

**Reply by saimyosett** [reply]
@worse435  @VenomXL I got the same problem and stuck around for days.
Please check all your config files and make sure that there is no space in front of the <?php  tag.

**Reply by VenomXL** [reply]
@saimyosett I am still stuck on this it just doesn't work it seems

**Reply by a4anthony** [reply]
@VenomXL Do you still have this issue?

**Reply by VenomXL** [reply]
@a4anthony Yes still facing the same

**Reply by a4anthony** [reply]
@VenomXL There's most likely a whitespace before <?php in one your files (like this " <?php"). If you can find and fix that it should work fine.

**Reply by tnt-leon-jashari** [reply]
have you managed to fix this? facing the same problem.
